evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -91,13 +91,36 @@ def test_data_splitter(X_y_binary):
     assert isinstance(automl.rankings, pd.DataFrame)
     assert len(automl.results['pipeline_results'][0]["cv_data"]) == cv_folds
 
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', data_splitter=TimeSeriesSplit(n_splits=cv_folds),
+                          max_iterations=1, n_jobs=1, engine="parallel")
+    automl.search()
+
+    assert isinstance(automl.rankings, pd.DataFrame)
+    assert len(automl.results['pipeline_results'][0]["cv_data"]) == cv_folds
+
 
 def test_max_iterations(X_y_binary):
     X, y = X_y_binary
     max_iterations = 5
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', max_iterations=max_iterations, n_jobs=1)
-    automl.search()
-    assert len(automl.full_rankings) == max_iterations
+
+    # Sequential
+    sequential_automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary',
+                                     max_iterations=max_iterations, n_jobs=1, engine="sequential")
+    sequential_automl.search()
+    assert len(sequential_automl.full_rankings) == max_iterations
+
+    # # Parallel
+    # parallel_automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary',
+    #                                max_iterations=max_iterations, n_jobs=1, engine="parallel")
+    # parallel_automl.search()
+    # assert len(parallel_automl.full_rankings) == max_iterations
+    #
+    # # Compare Sequential and Parallel Ranking Values
+    # sequential_rankings = sequential_automl.full_rankings
+    # sequential_rankings.set_index("id")
+    # parallel_rankings = parallel_automl.full_rankings
+    # parallel_rankings.set_index("id")
+    # assert all(parallel_rankings == sequential_rankings)
```

Comment:
Limiting the parallelengine to max_iterations needs further consideration.  If the intent of limiting max iterations was to reduce the runtime of the AutoMLSearch sequentially running through pipelines, then it shouldn't be a big deal that the search is just executing all the pipelines.

Reply:
So the current functionality that I left in is that it still works when max_iterations is passed, just ignored.  The callbacks are currently no bueno in their existing form as all the workers are processing pipelines in parallel, like you said, and should_continue() doesn't know when self._results or self._pipelines whatever it is has triggered the should_continue() condition.

Reply:
I think it also goes without saying that I'm going to delete this code comment, I just wanted to show everyone what I was trying to do.